arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/bookmark/BookmarkAdapter.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.toolkit.bookmark
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.esri.arcgisruntime.mapping.Bookmark
+import com.esri.arcgisruntime.toolkit.R
+import kotlinx.android.synthetic.main.item_bookmark_row.view.*
+
+class BookmarkAdapter(
+    private val onItemClickListener: OnItemClickListener<Bookmark>,
+    diffCallback: DiffUtil.ItemCallback<Bookmark> = DiffCallback(),
+    private val itemLayoutRes: Int? = null
```

Comment:
You can get rid of the null here and the null check further down by providing a default optional parameter in this constructor.

```
class BookmarkAdapter(
    private val onItemClickListener: OnItemClickListener<Bookmark>,
    @LayoutRes private val  itemLayoutRes: Int = R.layout.item_bookmark_row,
    diffCallback: DiffUtil.ItemCallback<Bookmark> = DiffCallback()
) : ListAdapter<Bookmark, ViewHolder>(diffCallback) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return ViewHolder(
            inflater.inflate(
                itemLayoutRes,
                parent,
                false
            )
        )
    }
```

And calling the constructor would look like:
```
bookmarksAdapter = BookmarkAdapter(
                object : BookmarkAdapter.OnItemClickListener<Bookmark> {
                    override fun onItemClick(item: Bookmark) {
                        onItemClickListener?.onItemClick(item)
                    }
                },
                itemLayoutRes = R.layout.your_own_layout
            )
```

Reply:
updated